src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,244 @@
+            }
+            if (chunkingAlgorithmCount > 1) {
+                throw new IllegalArgumentException("multiple chunking algorithms defined for input field [" + inputField + "]");
+            }
+        }
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument document) {
+        for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
+            String inputField = fieldMapEntry.getKey();
+            Object content = document.getFieldValue(inputField, Object.class);
+
+            if (content == null) {
+                throw new IllegalArgumentException("input field in document [" + inputField + "] is null, cannot process it.");
+            }
+
+            if (content instanceof List<?>) {
+                List<?> contentList = (List<?>) content;
+                for (Object contentElement : contentList) {
+                    if (!(contentElement instanceof String)) {
+                        throw new IllegalArgumentException(
+                            "some element in input field list ["
+                                + inputField
+                                + "] of type ["
+                                + contentElement.getClass().getName()
+                                + "] cannot be cast to ["
+                                + String.class.getName()
+                                + "]"
+                        );
+                    }
+                }
+            } else if (!(content instanceof String)) {
+                throw new IllegalArgumentException(
+                    "input field ["
+                        + inputField
+                        + "] of type ["
+                        + content.getClass().getName()
+                        + "] cannot be cast to ["
+                        + String.class.getName()
+                        + "]"
+                );
+            }
+
+            Map<?, ?> parameters = (Map<?, ?>) fieldMapEntry.getValue();
+            String outputField = (String) parameters.get(OUTPUT_FIELD);
+            List<String> chunkedPassages = new ArrayList<>();
+
+            // we have validated that there is one chunking algorithm
+            // and that chunkerParameters is of type Map<String, Object>
+            for (Map.Entry<?, ?> parameterEntry : parameters.entrySet()) {
+                String parameterKey = (String) parameterEntry.getKey();
+                if (supportedChunkers.contains(parameterKey)) {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> chunkerParameters = (Map<String, Object>) parameterEntry.getValue();
+                    if (Objects.equals(parameterKey, ChunkerFactory.FIXED_LENGTH_ALGORITHM)) {
+                        // add maxTokenCount to chunker parameters
+                        Map<String, Object> sourceAndMetadataMap = document.getSourceAndMetadata();
+                        int maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(settings);
```

Comment:
Why do we need to read this from settings?

Reply:
Because MAX_TOKEN_COUNT_SETTING is a setting related to index, which indicates the maximum number of tokens for tokenizer analyzer for this index.